CRM/Core/Payment.php

```diff
@@ -1181,6 +1227,11 @@ public function doPayment(&$params, $component = 'contribute') {
       return $result;
     }
 
+    // Set some "Standard" parameters for payment
+    $this->getIpAddress($params);
```

Comment:
I feel like calling this functions should be done in the payment processor classes - rather than here

Reply:
(I'm OK with adding the functions per above)

Reply:
@eileenmcnaughton why wouldn't this happen here, just thinking if these fields applied equally to all payment processors why wouldn't as core have it all in one place?

Reply:
@seamuslee001 well 
a) there is an existing pattern of processors using the get function when they want to get certain details (getAmount for example)
b) we actually encourage processors to entirely override doPayment these days
c) we are mixing get & set in the same action - ie. we have $params but we are doing a set within it calling get

Note that the issue here is that core CiviCRM actually ensures these things are set before calling doPayment (I don't know what gaps there are) as does webform - but the goal is to move that out of webform into these classes. Perhaps at some point also out of the core forms but really what we need is a much clearer 'contract' as to what is required when calling doPayment